app/controllers/comment_controller.php

```diff
@@ -11,11 +16,19 @@ public function view()
         $page = Param::get('page', 1);
         $pagination = new SimplePagination($page, self::MAX_ITEM_PER_PAGE);
 
-        $comments = $comment->getAll($pagination->start_index-1, $pagination->count+1, $thread->id);
+        $filter_username = htmlentities(Param::get('username'));
+
+        $comments = $comment->getAll($pagination->start_index-1, $pagination->count+1, $thread->id, $filter_username);
         $pagination->checkLastPage($comments);
 
-        foreach ($comments as $comment) {
-            $comment->user_id = User::getUsernameById($comment->user_id);
+        $comment->getUserAttributes($comments);
+
+        $sort = Param::get('sort');
+
+        if($sort === self::SORT_TYPE_COMMENT) {
+            usort($comments, function($a , $b) {
+                return $b->likecount - $a->likecount;
+            });
```

Comment:
Done po https://github.com/renzosunico/MyClassroom/commit/277564d0edce3944ac28361efb2b32e3028be1d5 :smile: Is it okay po if I use Param::get as parameter to a function or should I store it  po in a variable first? Thank you po. :bow: